scripts/cms-any/dispatch-ci.php

```diff
@@ -32,6 +32,14 @@
         uses: silverstripe/gha-dispatch-ci@v1
 EOT;
 
-if (check_file_exists('.github/workflows/ci.yml')) {
-    write_file_even_if_exists('.github/workflows/dispatch-ci.yml', $content);
+$dispatchCiPath = '.github/workflows/dispatch-ci.yml';
+$ciPath = '.github/workflows/ci.yml';
+$shouldHaveDispatchCi = (is_module() || is_composer_plugin()) && !is_docs() && !is_gha_repository();
```

Comment:
Realistically I think `is_module() || is_composer_plugin()` is all that's needed here, but doesn't hurt to be explicit so it can stay as is.